### PR TITLE
test: Set up integration tests with TestContainers and RestAssured

### PR DESCRIPTION
**Pull request by gspaiva**
## Description

- Add integration test structure to the project
- Add Test containers
- Add `-x test` parameter to the ./gradlew build to NOT run tests during the image build
- Add new endpoint on sample controller

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Integration test <3

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by felipetomm — APPROVED** [review]
Ok

**Review by felipetomm — APPROVED** [review]
Ok

File: infrastructure/src/main/java/postech/soat/tech/challenge/repository/CustomerRepository.java [hunk 1]
```diff
@@ -3,5 +3,5 @@
 import org.springframework.data.repository.CrudRepository;
 import postech.soat.tech.challenge.entity.Customer;
 
-public interface CustomerRepository extends CrudRepository<Customer, Long>{
+public interface CustomerRepository extends CrudRepository<Customer, Long> {
```
Comment (andreoliveiramorais) [inline infrastructure/src/main/java/postech/soat/tech/challenge/repository/CustomerRepository.java:6]: @gspaiva que tal esse repo ficar dentro de uma pasta persistence/repository? assim talvez fique mais organizado. o mesmo pra entity.

Comment (gspaiva) [reply infrastructure/src/main/java/postech/soat/tech/challenge/repository/CustomerRepository.java:6]: Ambos já estão, esse está dentro do pacote repository e a entidade no pacote entity.

File: infrastructure/src/main/java/postech/soat/tech/challenge/api/controller/SampleController.java [hunk 2]
```diff
@@ -27,7 +30,14 @@ public ApiResponse<String> sampleEndpoint() {
...
     @PostMapping("/customer")
-    public ApiResponse<Customer> inserirCliente(@RequestBody Customer customer) {
+    public ApiResponse<Customer> addCustomer(@RequestBody Customer customer) {
         return new ApiResponse<>(customerRepository.save(customer));
     }
+
+    @GetMapping("/customer/all")
+    public ApiResponse<List<Customer>> listCustomers() {
+        List<Customer> customers = new ArrayList<>();
+        customerRepository.findAll().forEach(customers::add);
```
[the remark this answers is not in the document]
Comment (gspaiva) [reply infrastructure/src/main/java/postech/soat/tech/challenge/api/controller/SampleController.java:40]: A interface está retornando um Iterator, não uma lista, aí preciso consumir tudo do iterator e passar para uma lista. Mas isso vai ser removido logo logo tb.

**Pull request merged**